python/test/test_palletjack.py

```diff
@@ -192,28 +198,30 @@ def test_index_file_golden_master(self):
             # Compare the actual output to the expected output
             self.assertEqual(actual_output, expected_output)
 
-        pr = pq.ParquetReader()
-        pr.open(path)
-        metadata = pr.metadata
-        row_groups = metadata.num_row_groups
-
-        for r in range(0, row_groups):
-
-            # Reading using the original metadata
             pr = pq.ParquetReader()
             pr.open(path)
-            res_data_org = pr.read_row_groups([r], use_threads=False)
+            metadata = pr.metadata
+            row_groups = metadata.num_row_groups
 
-            # Reading using the indexed metadata
-            metadata = pj.read_metadata(expected_index_path, row_groups = [r])
-            pr = pq.ParquetReader()
-            pr.open(path, metadata=metadata)
+            for r in range(0, row_groups):
+
+                # Reading using the original metadata
+                pr = pq.ParquetReader()
+                pr.open(path)
+                res_data_org = pr.read_row_groups([r], use_threads=False)
+
+                # Reading using the indexed metadata
+                metadata = pj.read_metadata(expected_index_path, row_groups = [r])
+                pr = pq.ParquetReader()
+                pr.open(path, metadata=metadata)
+
+                res_data_index = pr.read_row_groups([0], use_threads=False)
+                self.assertEqual(res_data_org, res_data_index, f"Row={r}")
+
+            pr.close()
```

Comment:
Should we be closing each Parquet reader before opening the next one, rather than only closing the one from the last iteration?